Modules/xASL_module_Population.m

```diff
@@ -265,25 +266,25 @@
             end
         end
     
-        % Check if we should do the same for Lesion or ROI masks (i.e.
-        % individual "atlases") -> NB not yet developed/tested in native space
-        LesionROIList = xASL_adm_GetFileList(x.D.PopDir, '(?i)^r(Lesion|ROI).*\.nii$', 'FPList', [0 Inf]);
+        % Check if we should do the same for Lesion or ROI masks (i.e. individual "atlases")
+		% PM: not yet developed/tested in native space
+        
+		% Read the names of the lesion files
+		LesionROIList = xASL_adm_GetFileList(x.D.PopDir, '(?i)^r(Lesion|ROI).*\.nii$', 'List', [0 Inf]);
+		% Go through the lesions and remove the subject names
+		for iLesion = 1:length(LesionROIList)

```

Comment:
Simplification:

[~, iEnd] = cellfun(@(y) regexp(y, '^r(Lesion|ROI)_(T1|FLAIR)_\d'), LesionROIList);

for iLesion = 1:length(LesionROIList)
    LesionROIList = LesionROIList{iLesion}(1:iEnd);
end

Reply:
Done partly. Cause in your example, there would be a variable iEnd for different atlas types.

Reply:
Yes sorry, should have been:

LesionROIList = LesionROIList{iLesion}(1:iEnd(iLesion));

Reply:
I did some tests and it crashed with some empty ROIs, so going through them 1-by-1 seems to be easier to debug.